pkg/cluster/kubeclient/backup.go

```diff
@@ -0,0 +1,64 @@
+package kubeclient
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"time"
+
+	batchv1 "k8s.io/api/batch/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/openshift/openshift-azure/pkg/util/ready"
+)
+
+const (
+	etcdBackupNamespace = "openshift-etcd"
+)
+
+var (
+	rxRfc1123 = regexp.MustCompile(`(?i)^` +
+		`([a-z0-9]|[a-z0-9][-a-z0-9]{0,61}[a-z0-9])` +
+		`(\.([a-z0-9]|[a-z0-9][-a-z0-9]{0,61}[a-z0-9]))*` +
+		`$`)
+)
+
+func (u *kubeclient) BackupCluster(ctx context.Context, backupName string) error {
+	u.log.Infof("running an etcd backup")
+	if len(backupName) > 255 || !rxRfc1123.MatchString(backupName) {
+		return fmt.Errorf("invalid backup name: %s", backupName)
+	}
+	cronjob, err := u.client.BatchV1beta1().CronJobs(etcdBackupNamespace).Get("etcd-backup", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	job := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      backupName,
+			Namespace: etcdBackupNamespace,
+		},
+		Spec: cronjob.Spec.JobTemplate.Spec,
+	}
+	job.Spec.Template.Spec.Containers[0].Args = []string{fmt.Sprintf("-blobname=%s", backupName), "save"}
+	backup, err := u.client.BatchV1().Jobs(etcdBackupNamespace).Create(job)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		int64Ptr := func(i int64) *int64 { return &i }
+		err = u.client.BatchV1().Jobs(etcdBackupNamespace).Delete(backupName, &metav1.DeleteOptions{GracePeriodSeconds: int64Ptr(0)})
```

Comment:
do you actually need to set GracePeriodSeconds?

Reply:
not really. i'll take it out